sigopt/xgboost/experiment.py

```diff
@@ -0,0 +1,82 @@
+from .run import run as XGBRun
+from .run import parse_run_options, PARAMS_LOGGED_AS_METADATA, DEFAULT_EVALS_NAME
+import copy
+
+from .. import create_experiment
+
+DEFAULT_CLASSIFICATION_METRICS = ['accuracy', 'precision', 'recall', 'F1']
+DEFAULT_REGRESSION_METRICS = ['mean absolute error', 'mean squared error']
+DEFAULT_NUM_BOOST_ROUND = 10
+
+
+class XGBExperiment:
+  def __init__(self, experiment_config, dtrain, evals, params, num_boost_round, run_options):
+    self.experiment_config = experiment_config
+    self.dtrain = dtrain
+    self.evals = evals
+    self.params = params
+    self.num_boost_round = num_boost_round
+    self.run_options = run_options
+    self.sigopt_experiment = None
+
+  def parse_and_create_parameter_space(self):
+    # TODO: deal with later
+    pass
+
+  def parse_and_create_experiment(self):
+    experiment_config_parsed = copy.deepcopy(self.experiment_config)
+
+    # Check experiment config optimization metric
+    for metric in experiment_config_parsed['metrics']:
+      if metric['strategy'] == 'optimize':
+        assert metric['name'] in DEFAULT_CLASSIFICATION_METRICS or metric['name'] in DEFAULT_REGRESSION_METRICS
+
+        # change optimized metric to reflect updated name
+        metric['name'] = self.evals[0][1] + '-' + metric['name'] if isinstance(self.evals, list) else \
+          DEFAULT_EVALS_NAME + '-' + metric['name']
```

Comment:
syntax, we don't do that `\` for line break. Just change this to a regular if block
```
if isinstance(...):
  metric['name'].......
else:
  ...
```